diff --git a/includes/saml_sp.AuthnRequest.inc b/includes/saml_sp.AuthnRequest.inc
index 8918bdc..29544ee 100644
--- a/includes/saml_sp.AuthnRequest.inc
+++ b/includes/saml_sp.AuthnRequest.inc
@@ -73,8 +73,6 @@ PROVIDERNAME;
     <samlp:NameIDPolicy
         Format="{$nameIDPolicyFormat}"
         AllowCreate="true" />
-    <samlp:RequestedAuthnContext Comparison="exact">{$authncontext}
-    </samlp:RequestedAuthnContext>
 </samlp:AuthnRequest>
 AUTHNREQUEST;
 
